orbstation/code/antagonists/contract_killer.dm

```diff
@@ -0,0 +1,223 @@
+/datum/antagonist/contract_killer
+	name = "Contract Killer"
+	show_in_antagpanel = TRUE
+	antagpanel_category = "Other"
+	job_rank = ROLE_CONTRACT_KILLER
+	antag_hud_name = "contractkiller"
+	ui_name = "AntagInfoMinor"
+	show_name_in_check_antagonists = TRUE
+	roundend_category = "contract killers"
+	count_against_dynamic_roll_chance = FALSE
+	silent = FALSE
+	suicide_cry = "FOR THE PAY!! Wait-"
+	preview_outfit = /datum/outfit/contract_killer
+
+/datum/antagonist/contract_killer/on_gain()
+	forge_objectives()
+	return ..()
+
+/datum/antagonist/contract_killer/greet()
+	owner.current.playsound_local(get_turf(owner.current), 'orbstation/sound/antag_alerts/contractkilleralert.ogg', 100, FALSE, pressure_affected = FALSE, use_reverb = FALSE)
+	owner.announce_objectives()
+
+/datum/antagonist/contract_killer/forge_objectives()
+	var/list/objectives_left = list("intel", "polaroid")
+	var/datum/objective/assassinate/kill_objective = new
+	kill_objective.owner = owner
+	var/datum/mind/target_mind = kill_objective.find_target()
+	var/obj/family_heirloom
+
+	//if the target has an heirloom, the contract killer might try to steal it
+	var/datum/quirk/item_quirk/family_heirloom/heirloom_quirk = locate() in target_mind.current.quirks
+	if(heirloom_quirk)
+		family_heirloom = heirloom_quirk.heirloom?.resolve()
+		if(family_heirloom)
+			objectives_left += "heirloom"
+
+	// If the target is not the captain, someone in the same department might also be a kill objective.
+	if(!is_captain_job(target_mind.assigned_role))
+		objectives_left += "coverup"
+
+	for(var/i in 1 to 2)
+		var/chosen_objective = pick(objectives_left)
+		objectives_left.Remove(chosen_objective)
+		switch(chosen_objective)
+			if("intel")
+				var/datum/objective/gather_intel/intel = new
+				intel.owner = owner
+				intel.target = target_mind
+				intel.start_ticking()
+				objectives += intel
+			if("polaroid")
+				var/datum/objective/polaroid/polaroid = new
+				polaroid.owner = owner
+				polaroid.target = target_mind
+				objectives += polaroid
+			if("heirloom")
+				var/datum/objective/steal/heirloom_thief/heirloom_thief = new
+				heirloom_thief.owner = owner
+				heirloom_thief.target = target_mind //while you usually wouldn't need this for stealing, we need the name of the obsession
+				heirloom_thief.steal_target = family_heirloom
+				objectives += heirloom_thief
+			if("coverup")
+				var/datum/objective/assassinate/jealous/jealous = new
+				jealous.owner = owner
+				jealous.target = target_mind //will reroll into a coworker on the objective itself
+				objectives += jealous
+
+	objectives += kill_objective //Add the assassinate last, because you're meant to save it for last.
+
+	//last objective
+	if(prob(70)) //70% - "clean kill" objective
+		var/datum/objective/clean_kill/clean = new
+		clean.owner = owner
+		objectives += clean
+	else if(prob(50)) //15% - "frame job" objective
```

Comment:
If anyone ever extends this to add a fourth option im asking them to just generate a random number between 1 and 100 but you're getting away with it today